### DISCUSSION TITLE
Save jason after every change.

### DISCUSSION THREAD

**Discussion by poleguy** [ckipp01/skan#99] [Feature Request]
It seems the .json on disk is only saved when you quit the program. It would seem less risky to save out the .json after every change. If the terminal accidentally gets killed we don't want to lose a bunch of items we just added.

**Comment by ckipp01** [comment]
Thanks for bringing this up. I'm a bit torn as this is a lot of writes. For myself I don't normally leave skan open for long periods of time, but rather just open and then do something and close. So I guess this is catered to my workflow a bit more than anything else.

**Comment by poleguy** [comment]
Is the concern that adding write calls to the code will be a lot of work or make the code messy? Or is the concern that you will wear out your ssd? Or something else? 

I feel a todo app's primary job is saving notes. So the combination of not saving till you close with a screen resizing bug that triggers a crash and leads to a loss of notes means it failed in its primary job. 

Also since it can't edit the notes I noticed it wasn't saving immediately when I opened the json file in an editor and it was stale.

If you decide not to save asap, then fix the crashes, add note editing, and then the only people who will notice is those who accidentally kill the app and lose work. 

If you can make it gracefully handle saving when you receive a kill signal then that's not a worry.
